Gdk-3.0.gir

```diff
@@ -22654,8 +22654,8 @@ public API and should not be used in application code.</doc>
           <doc xml:space="preserve">the data (a `guchar *`
   `gushort *`, or `gulong *`,
   depending on @format), cast to a `guchar *`.</doc>
-          <type name="guint8" c:type="const guchar*"/>
-        </parameter>
+          
+        <array length="6" zero-terminated="0" c:type="const guint8*"><type name="guint8" c:type="guint8"/></array></parameter>
```

Comment:
I don't think this is correct.

The docs say `the data (a guchar * gushort *, or gulong *, depending on format ), cast to a guchar *`

A `gulong` could be 8 bytes, and does it mean a pointer to a single `guchar`/etc or a pointer to `nelements` of them? I understand the latter, so this could be arbitrary size.

Reply:
Considering all that, I assume the safe bindings to this function in GDK are unsafe and broken :)

Reply:
SO should we change the type or the length?

Reply:
@sdroege You worried about `length=6` ? It just number of parameter with length: "nelements" if I count right.

Reply:
I can do that.

Reply:
Yes. `nelements` is the number of elements of the given type, not the number of bytes

Note that `gulong` is super-annoying. You have to use `libc::c_ulong` for this in the API. And I would suggest using manual bindings that use an enum in the API.

```rust
enum Whatever {
  UChars(&[u8]),
  UShorts(&[u16]),
  ULongs(&[libc::c_ulong]),
}
```

Reply:
IMHO need also cases for single value, as property usually has single value and array cases is rare.

Reply:
Single values: `Whatever::UChars(&[123u8])` :)

Are the uchars usually literally numbers or often UTF8/ASCII strings?

Reply:
I meant: Long(libc::c_long), Int (libc::c_int) etc. passed with nelements=1 it I understand right.
Not sure about string.